pkg/sources/source_manager.go

```diff
@@ -511,6 +591,44 @@ func (s *SourceManager) runWithUnits(ctx context.Context, source SourceUnitEnumC
 	}
 }
 
+// scanWithUnits produces chunks from a single SourceUnit.
+func (s *SourceManager) scanWithUnits(ctx context.Context, source SourceUnitChunker, report *JobProgress, unit SourceUnit) error {
```

Comment:
This method name confuses me - it refers to unit**s** (plural), but only scans one unit, right?

Reply:
correct. updated